Source/WebCore/html/TextFieldInputType.h

```diff
@@ -121,7 +121,8 @@ class TextFieldInputType : public InputType, protected SpinButtonElement::SpinBu
     bool shouldDrawCapsLockIndicator() const;
     bool shouldDrawAutoFillButton() const;
 
-    void createContainer();
+    enum class PreserveSelectionRange : bool { No, Yes };
+    void createContainer(PreserveSelectionRange = PreserveSelectionRange::Yes);
```

Comment:
I think it makes more sense for the default to be no, that's the safest option. Ideally restoring the selection range should be separate from shadow root container creation

Reply:
The idea is that any other call to `createContainer` would be to add a decoration, in which case the selection does need to be preserved.

I also think this change is much safer, since the restoration occurs asynchronously, and we have a `WeakPtr` to the element. `PreserveSelectionRange` is more about avoiding unnecessary work at this point.